src/Razor/test/Microsoft.CodeAnalysis.Razor.Workspaces.Test/Serialization/TagHelperDescriptorSerializationTest.cs

```diff
@@ -28,98 +24,44 @@ public TagHelperDescriptorSerializationTest(ITestOutputHelper testOutput)
     public void TagHelperDescriptor_DefaultBlazorServerProject_RoundTrips()
     {
         // Arrange
-        var testFileName = "test.taghelpers.json";
-        var current = new DirectoryInfo(AppContext.BaseDirectory);
-        while (current != null && !File.Exists(Path.Combine(current.FullName, testFileName)))
-        {
-            current = current.Parent;
-        }
-
-        var tagHelperFilePath = Path.Combine(current.FullName, testFileName);
-        var buffer = File.ReadAllBytes(tagHelperFilePath);
-        var serializer = new JsonSerializer();
-        serializer.Converters.Add(TagHelperDescriptorJsonConverter.Instance);
-        IReadOnlyList<TagHelperDescriptor> deserializedTagHelpers;
+        var bytes = TestResources.GetResourceBytes(TestResources.BlazorServerAppTagHelpersJson);
 
         // Act
-        using (var stream = new MemoryStream(buffer))
-        using (var reader = new JsonTextReader(new StreamReader(stream)))
-        {
-            deserializedTagHelpers = serializer.Deserialize<IReadOnlyList<TagHelperDescriptor>>(reader);
-        }
 
-        MemoryStream serializedStream;
-        using (serializedStream = new MemoryStream())
-        using (var writer = new StreamWriter(serializedStream, Encoding.UTF8, bufferSize: 4096))
+        // Read in the tag helpers
+        IReadOnlyList<TagHelperDescriptor> expectedTagHelpers;
+
+        using (var stream = new MemoryStream(bytes))
+        using (var reader = new StreamReader(stream))
         {
-            serializer.Serialize(writer, deserializedTagHelpers);
+            expectedTagHelpers = JsonDataConvert.DeserializeData(reader,
+                static r => r.ReadArrayOrEmpty(
+                    static r => ObjectReaders.ReadTagHelper(r, useCache: false)));
         }
 
-        IReadOnlyList<TagHelperDescriptor> reDeserializedTagHelpers;
-        var reserializedStream = new MemoryStream(serializedStream.GetBuffer());
-        using (reserializedStream)
-        using (var reader = new JsonTextReader(new StreamReader(reserializedStream)))
+        using var writeStream = new MemoryStream();
+
+        // Serialize the tag helpers to a stream
+        using (var writer = new StreamWriter(writeStream, Encoding.UTF8, bufferSize: 4096, leaveOpen: true))
         {
-            reDeserializedTagHelpers = serializer.Deserialize<IReadOnlyList<TagHelperDescriptor>>(reader);
+            JsonDataConvert.SerializeData(writer,
+                r => r.WriteArray(expectedTagHelpers, ObjectWriters.Write));
         }
 
-        // Assert
-        Assert.Equal(deserializedTagHelpers, reDeserializedTagHelpers);
-    }
+        // Deserialize the tag helpers from the stream we just serialized to.
+        writeStream.Seek(0, SeekOrigin.Begin);
 
-    [Fact]
-    public void TagHelperDescriptor_CanReadCamelCasedData()
```

Comment:
I removed this test, because it did not do what it said it did. It appeared to be trying to test JSON serialization when property names are camel-cased. However, the code doesn't actually do that. Simply including a `CamelCasePropertyNamesContractResolver` when serializing an object to JSON does not automatically camel-case the property names if the JSON converters specifically write pascal-cased names. In addition, there shouldn't be a reason for us to care about casing of property names. All of our serialization is machine-produced and machine-consumed.